### PR TITLE
Scope 'push' GitHub Actions workflow trigger to 'main' branch

### PR DESCRIPTION
**Pull request by lopopolo**
The previous configuration causes all jobs to run twice on every PR. We should consider protecting the main branch and requiring PRs if that is not already the case.

### CODE REVIEW

**Review by astraw — APPROVED** [review]

**Comment by astraw** [comment]
The CI failures on Android look like a nightly rust problem as they are not happening on stable nor on nightly on other platforms. The Mac issue is described at #72 and is also unrelated.

I don't understand as much about the github CI admin config as I could, but I'll take your word for it that this is a good thing.

In a similar vein, I just did as you suggested and created a new rule that a PR is required before merging to `main` and a require also minimum of 1 approval.

Relatedly, I made you and @Kijewski both admins on this repo, so we now mostly have a bus factor of 3. Also at the moment I think only @Kijewski can publish releases to crates.io. Happy to also add you there if you like, too.

**Comment by lopopolo** [comment]
Thanks for making those changes @astraw. I think I'd be more comfortable without publisher privileges for now if that's ok. If there's a need to add another publisher later, we can revisit that though.

**Comment by lopopolo** [comment]
@astraw I believe https://github.com/rust-lang/cc-rs/pull/728 will fix the issues with android on nightly since the latest nightly made changes to which Android NDK is used: https://rust-lang.zulipchat.com/#narrow/stream/219381-t-libs/topic/1.2E65.20and.20the.20Android.20NDK

**Pull request merged**
